vllm/v1/engine/llm_engine.py

```diff
@@ -155,6 +155,12 @@ def __init__(
         # GPU and CPU blocks, which are profiled in the distributed executor.
         self.scheduler = Scheduler(scheduler_config, cache_config, lora_config)
 
+    def __del__(self):
+        # Small hack- implicit clean up of resources on garbage collect
+        # TODO: this should probably be explicitly invoked when we're done with
+        # the engine
+        self.terminate_detokenizer()
```

Comment:
This is being reworked anyhow right now.

Reply:
I was hoping as much!